### PR TITLE
DB_Schema update:  Data_source table, PSHTT, and combined credential domain alerts tables

### LINKED ISSUE #107: Add Table Called DataSource_Table to P&E Schema
# 💡 Summary #

The P&E database needs to have a table dedicated to data sources.

## Motivation and context ##

There are duplicative tables that are based on slight differences in data sources.  (ex: CyberSixGill Credentials vs. HIBP Credentials)

This will reduce the complexity of the database.

## Implementation notes ##

Create a table called Datasource_tbl with the following fields:

- dsource_uuid
- dsource_vendorname
- dsource_supportemail
- dsource_url
- dsource_api_url
- dsource_api_limits

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Data source table will exist
- [ ] Data source table is populated with relevant data

### LINKED ISSUE #110: Modify Database Tables to Use DataSource Table
# 💡 Summary #

All Database tables will be modified to use the datasource uuid rather than tables named for the datasources.

## Motivation and context ##

This is to reduce complexity of database.

## Implementation notes ##

Please provide details for implementation, such as:

- Database tables will be modified to use the datasource uuid field
- Redundant tables will be combined into central tables based on common data.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] No more tables will be named after the datasource
- [ ] Database tables will attribute data to the source

### PR DESCRIPTION
**Pull request by aloftus23**
## 🗣 Description ##
Added data_source table and got rid of any source specific tables. For example, added a data_source column to the "credential_exposures" table to specify the source: either cybersix and hibp (instead of a hibp_exposed_credential table)

Added PSHTT table.
fixes #107 
fixes #110 
## 💭 Motivation and context ##

Normalize the db schema.

## 🧪 Testing ##

Successfully built the database using this schema.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: src/pe_reports/data/data_schema.sql [hunk 1]
```diff
@@ -1,138 +1,226 @@
...
+--
+
+CREATE TABLE public.credential_breaches (
+    credential_breaches_uid uuid DEFAULT public.uuid_generate_v1() NOT NULL,
+    breach_name text NOT NULL,
+    description text,
+    exposed_cred_count bigint,
+    breach_date date,
+    added_date timestamp without time zone,
+    modified_date timestamp without time zone,
```
Comment (dav3r) [inline src/pe_reports/data/data_schema.sql]: What is the reason for storing timestamps without a time zone?  Generally, that will get you into trouble at some point.

**Review by dav3r — APPROVED** [review]
Looks fine to me, though I did have one question about timestamps.

File: src/pe_reports/data/data_schema.sql (hunk 1 above)
Comment (aloftus23) [reply src/pe_reports/data/data_schema.sql]: That is a very good catch. Those columns are set to just timestamp in our database but the schema-dump, using pg_dump, changed it for some reason. I'm going to make those adjustments and double-check the rest.

Comment (dav3r) [reply src/pe_reports/data/data_schema.sql]: Looks like you remedied this in https://github.com/cisagov/pe-reports/pull/126/commits/8564cb8b0300091c3e7da1bb918d87aa65ca5dcc. 👍

File: src/pe_reports/data/data_schema.sql [hunk 2]
```diff
@@ -168,27 +468,29 @@ CREATE TABLE IF NOT EXISTS public.shodan_insecure_protocols_unverified_vulns
...
+
+CREATE TABLE public.shodan_verified_vulns (
+    verified_vuln_uid uuid DEFAULT public.uuid_generate_v1() NOT NULL,
     organizations_uid uuid NOT NULL,
     organization text,
     ip text,
     port text,
     protocol text,
-    timestamp timestamp,
+    "timestamp" timestamp,
```
Comment (cduhn17) [inline src/pe_reports/data/data_schema.sql]:
```suggestion
    event_time timestamp,
```

**Review by cduhn17 — CHANGES_REQUESTED** [review]
@aloftus23 
One small adjustment to the column name change at shodan_verified_vulns.

**Review by cduhn17 — APPROVED** [review]

**Comment by cduhn17** [comment]
Thank you everyone for your continued diligence and support in moving PR's along!

**Pull request merged**